color/quoted.h

```diff
@@ -29,16 +29,40 @@
 #include "core/lib.h"
 #include "color.h"
 
+struct Buffer;
+
 /// Ten colours, quoted0..quoted9 (quoted and quoted0 are equivalent)
 #define COLOR_QUOTES_MAX 10
 
 extern int QuotedColors[];
 extern int NumQuotedColors;
 
+/**
+ * struct QClass - Style of quoted text
+ *
+ * NeoMutt will store a tree of all the different email quoting levels it
+ * detects in an Email.  If $quote_regex matches, say both "> " and "| ",
+ * and the Email has three levels of indent, then the tree will contain two
+ * siblings each with a child and grandchild.
+ */
+struct QClass
+{
+  int quote_n;                  ///< The quoteN colour index for this level
+  int color;                    ///< Curses colour pair
+  char *prefix;                 ///< Prefix string, e.g. "> "
+  size_t prefix_len;            ///< Length of the prefix string
+  struct QClass *prev, *next;   ///< Different quoting styles at the same level
```

Comment:
Can we provide any example in the docstring? I'm not 100% sure what this means. 
Is it `> > > Some quote` where `prev` and `next` would one of the `>`s?

Reply:
I've added some GraphViz to the docstring.
It'll get rendered as something like:
![image](https://user-images.githubusercontent.com/76760/139845871-2d28e741-aa01-4e06-8e42-5e8e2b2f4d47.png)

Reply:
What does that picture mean? I don't understand what it's trying to convey.

Reply:
That's the tree of `QuoteStyle` objects representing the quoted lines found in the email.

It's a complicated example, but it represents and email with many replies and two styles of reply-indent:
- People replying by adding `> `
- People replying by adding `| `

As there have been many replies to replies, some lines have multiple `> ` indents (and some multiple `| ` indents).

Each QuoteStyle (blue box) stores the pattern to match, e.g. `> >`, and has a `color` that the text should be displayed.

Reply:
So the fields are to different `prefixes` at the same quote depth?

Reply:
yes